public/components/trace_analytics/components/dashboard/mode_picker.tsx

```diff
@@ -13,14 +13,14 @@ const labels = new Map([
 ]);
 
 export function DataSourcePicker(props: {
-  modes: {
+  modes: Array<{
     id: string;
     title: string;
-  }[];
+  }>;
   selectedMode: TraceAnalyticsMode;
   setMode: (mode: TraceAnalyticsMode) => void;
 }) {
-  const { modes, selectedMode, setMode } = props;
+  const { modes = [], selectedMode, setMode } = props;
```

Comment:
any reason modes is empty here? or is it a linting issue?

Reply:
Mode is not selected when entering from applications, needs to handle the empty state properly for it to render.